src/Sulu/Bundle/MediaBundle/Resources/config/services.xml

```diff
@@ -323,6 +324,7 @@
 
             <tag name="sulu.content.type" alias="single_media_selection"/>
             <tag name="sulu.content.export" format="1.2.xliff" translate="false" />
+            <tag name="sulu_admin.property_metadata_mapper" type="single_media_selection"/>
```

Comment:
The `type="single_media_selection"` is something I personally would not longer do as it make an interface not be able to be autoconfigureable I would add a static `getType` to the Interface instead this make it able to autowire/autoconfigure the Interface and no service definition need to be created when used in a project.

Reply:
But because those `PropertyMetadataMapper`s are mostly implemented in their corresponding content types, that may lead to confusion or even naming conflicts, because most content types implement quite a lot interfaces

Reply:
havent we added a `default-index-method` like discussed in https://github.com/sulu/sulu/pull/5640#discussion_r529658444? 🤔  
if i remember right, you dont need to implement the method if you set the index explicitly on the service

Reply:
You two seemed to agree in that comment, that's why I thought it is OK as it is now :see_no_evil: But apart from that, you can still add a static `getDefaultIndexName` method now, and it will work with autoconfiguring, if I am not mistaken.

Reply:
I thought, it would only make sense, if that method is also declared in the interface ... But I have to touch those files again very soon to implement a `PropertyMetadataMapper` for `selection` and `single_selection` field types, so I can add the `default-index-method` there without declaring the corresponding method in the `PropertyMetadataMapperInterface`

Reply:
you are right, we cannot add it to the `PropertyMetadataMapperInterface` and it looks like adding a `getDefaultIndexName` should already work: https://symfony.com/doc/current/service_container/tags.html#tagged-services-with-index

dont have a strong opioniern on adding am explicit `default-index-method` 🙂

Reply:
Well, I thought you *can* add a static method to the `PropertyMetadataMapperInterface`, but then you also force people to implement it, although they might want to register the service manually. Don't know how often that's the case, and the currently implemented variant works for me too.